src/components/templates/PosterHall/PosterHall.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+
+import { GenericVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+import { enterVenue } from "utils/url";
+
+import { usePosters } from "hooks/posters";
+
+import { PosterPreview } from "./components/PosterPreview";
+import { Search } from "./components/Search";
+
+import "./PosterHall.scss";
+
+export interface PosterHallProps {
+  venue: WithId<GenericVenue>;
+}
+
+export const PosterHall: React.FC<PosterHallProps> = ({ venue }) => {
+  const {
+    posterVenues,
+    titleFilter,
+    setTitleFilter,
+    liveFilter,
+    setLiveFilter,
+  } = usePosters({
+    posterHallId: venue.id,
+  });
+
+  return (
+    <div className="posterhall">
+      <Search
+        setTitleValue={setTitleFilter}
+        titleFilterValue={titleFilter}
+        liveFilterValue={liveFilter}
+        setLiveValue={setLiveFilter}
+      />
+
+      <div className="posterhall__posters">
+        {posterVenues?.map((posterVenue) => (
+          <PosterPreview
+            posterVenue={posterVenue}
+            key={posterVenue.id}
+            onClick={() => enterVenue(posterVenue.id)}
+          />
+        ))}
```

Comment:
`useMemo` (maybe called `renderedPosterVenues`)

Reply:
Ideally also make `key` the first prop.

Reply:
Fixed in https://github.com/sparkletown/sparkle/pull/1299/commits/141ddd81e6429c1fb2b7db9dffef1c62487c52f1